app/scripts/directives/processTemplateDialog.js

```diff
@@ -124,5 +234,82 @@
     function instantiateTemplate() {
       $scope.$broadcast('instantiateTemplate');
     }
+
+    function filterForKeywords (searchText, items) {
+      return KeywordService.filterForKeywords(items, ['name', 'tags'], KeywordService.generateKeywords(searchText));
+    }
+
+    function filterChange (filters) {
+      // only use filters which have a filter criteria 'value'
+      // prevents applying an empty keyword filter
+      // TODO: can remove the following line of code after angular-patternfly issue is fixed:
+      //   https://github.com/patternfly/angular-patternfly/issues/509
+      filters = _.filter(filters, 'value');
+
+      ctrl.filterConfig.appliedFilters = filters;
+      filterItems();
+    }
+
+    function filterItems() {
+      ctrl.filteredItems = ctrl.catalogItems;
+      if (ctrl.filterConfig.appliedFilters && ctrl.filterConfig.appliedFilters.length > 0) {
+        _.each(ctrl.filterConfig.appliedFilters, function(filter) {
+          ctrl.filteredItems = filterForKeywords(filter.value, ctrl.filteredItems);
+        });
+      }
+
```

Comment:
Curious if order of flow here matters, since what is inside `updateFilterControls()` is wrapped in `$timeout`...ie, could this call be the last line in this fn since its contexts will execute later?

Reply:
Adjusted, not sure what you were getting at but the timeout was unnecessary.

Reply:
Great!  I just have a tendency to pause and ask a question when I see `$timeout`, they are good at hiding bugs.  Wasn't sure about your code here, but figured I'd mention just in case.